test/AutoDiff/validation-test/simple_math.swift

```diff
@@ -121,6 +121,32 @@ SimpleMathTests.test("MultipleResults") {
   expectEqual((4, 3), gradient(at: 3, 4, of: multiply_swapAndReturnProduct))
 }
 
+// Test function with multiple `inout` parameters and a custom pullback.
```

Comment:
Did you mention that functions returning direct tuples of `Differentiable` values can also be differentiated? If so, could you please add a test case?

Reply:
Great suggestion, I added tests for returning tuples, and that uncovered a result type issue I hadn't considered. It took me a little bit to track it down, but the tests and a fix for this have been added in the latest commit.